src/cascadia/TerminalApp/TitlebarControl.cpp

```diff
@@ -0,0 +1,99 @@
+﻿//
+// TitlebarControl.xaml.cpp
+// Implementation of the TitlebarControl class
+//
+
+#include "pch.h"
+
+#include "TitlebarControl.h"
+
+#include "TitlebarControl.g.cpp"
+
+namespace winrt::TerminalApp::implementation
+{
+    TitlebarControl::TitlebarControl()
+    {
+        InitializeComponent();
+
+        // Register our event handlers on the MMC buttons.
+        MinMaxCloseControl().MinimizeClick({ this, &TitlebarControl::Minimize_Click });
+        MinMaxCloseControl().MaximizeClick({ this, &TitlebarControl::Maximize_Click });
+        MinMaxCloseControl().CloseClick({ this, &TitlebarControl::Close_Click });
+    }
+
+    Windows::UI::Xaml::UIElement TitlebarControl::Content()
+    {
+        return ContentRoot().Children().Size() > 0 ? ContentRoot().Children().GetAt(0) : nullptr;
+    }
+
+    void TitlebarControl::Content(Windows::UI::Xaml::UIElement content)
+    {
+        ContentRoot().Children().Clear();
+        ContentRoot().Children().Append(content);
+    }
+
+    void TitlebarControl::Root_SizeChanged(const IInspectable& sender, Windows::UI::Xaml::SizeChangedEventArgs const& e)
+    {
+        const auto windowWidth = ActualWidth();
+        const auto minMaxCloseWidth = MinMaxCloseControl().ActualWidth();
+        const auto dragBarMinWidth = DragBar().MinWidth();
+        const auto maxWidth = windowWidth - minMaxCloseWidth - dragBarMinWidth;
+        ContentRoot().MaxWidth(maxWidth);
+    }
+
+    uint64_t TitlebarControl::ParentWindowHandle() const
+    {
+        return reinterpret_cast<uint64_t>(_window);
+    }
+
+    void TitlebarControl::ParentWindowHandle(uint64_t handle)
+    {
+        _window = reinterpret_cast<HWND>(handle);
+    }
+
+    void TitlebarControl::_OnMaximizeOrRestore(byte flag)
+    {
+        if (_window)

```

Comment:
This is weird. Shouldn't it be an error or invalid to be dispatching events against one of these that doesn't have an attached window handle? Why is it just skipped?

Reply:
Oooh that is a good point. I just copied that from the MixMaxCloseControl. The TitlebarControl should _always_ have a `_window`.